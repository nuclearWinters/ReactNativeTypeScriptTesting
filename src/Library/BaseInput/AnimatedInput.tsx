import React from 'react';
import BaseInput, {Props} from './BaseInput';
import {StyleSheet} from 'react-native';

const defaultStyles = StyleSheet.create({
  box: {
    marginHorizontal: 40,
  },
  borderBottom: {
    backgroundColor: 'rgb(0,100,180)',
  },
  textInput: {
    borderBottomWidth: 2,
    borderBottomColor: 'lightgray',
    fontSize: 18,
    color: 'black',
  },
});

const {box, borderBottom, textInput} = defaultStyles;

const AnimatedInput: React.FC<Props> = ({
  placeholder,
  placeholderColor,
  onChangeText,
  value,
  styles,
}) => (
  <BaseInput
    onChangeText={onChangeText}
    placeholderColor={placeholderColor}
    placeholder={placeholder}
    value={value}
    styles={{
      box: [box, styles?.box],
      borderBottom: [borderBottom, styles?.borderBottom],
      textInput: [textInput, styles?.textInput],
    }}
  />
);

export default AnimatedInput;
