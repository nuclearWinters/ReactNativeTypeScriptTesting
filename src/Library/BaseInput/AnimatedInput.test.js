import React from 'react';
import {fireEvent, render, wait} from '@testing-library/react-native';
import AnimatedInput from './AnimatedInput';

test('AnimatedInput', async () => {
  const placeholder = 'palceholder test';
  let value = 'value test';
  const onChangeText = (text) => {
    value = text;
  };
  const placeholderColor = 'black';
  const {getByTestId, getByText, queryByTestId, baseElement, rerender} = render(
    <AnimatedInput
      placeholderColor={placeholderColor}
      placeholder={placeholder}
      value={value}
      onChangeText={onChangeText}
    />,
  );

  expect(getByTestId('textInput').props.value).toBe(value);
  const input = getByTestId('textInput');
  fireEvent.focus(input);
  await wait(() =>
    expect(getByTestId('borderBottom').props.style.left).toBe('0%'),
  );
  fireEvent.change(input, {nativeEvent: {text: value + '!'}});
  fireEvent.blur(input);
  await wait(() =>
    expect(getByTestId('borderBottom').props.style.left).toBe('50%'),
  );
  expect(getByTestId('textInput').props.value).toBe(value);
  expect(baseElement).toMatchSnapshot();
});
