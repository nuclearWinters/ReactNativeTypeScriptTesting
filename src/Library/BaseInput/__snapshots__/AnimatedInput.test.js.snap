// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedInput 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {
              "marginHorizontal": 40,
            },
            undefined,
          ],
        ]
      }
      testID="box"
    >
      <TextInput
        allowFontScaling={true}
        placeholder="palceholder test"
        placeholderTextColor="gray"
        rejectResponderTermination={true}
        style={
          Array [
            Array [
              Object {
                "borderBottomColor": "lightgray",
                "borderBottomWidth": 2,
                "color": "black",
                "fontSize": 18,
              },
              undefined,
            ],
          ]
        }
        testID="textInput"
        underlineColorAndroid="transparent"
        value="value test"
      />
      <View
        style={
          Object {
            "backgroundColor": "rgb(0,100,180)",
            "bottom": 0,
            "height": 2,
            "left": "50%",
            "position": "absolute",
            "right": "50%",
          }
        }
        testID="borderBottom"
      />
    </View>
  </View>
</View>
`;
